javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,371 @@
+'use strict';
+
+const { last } = require("cheerio/lib/api/traversing");
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that iterates over an array of people objects
+and creates a new list of each person's full name using the array method 'map'.
+Each object will have the shape {firstName:string, lastName:string}
+E.g. [ { firstName:"Jane", lastName:"Doe" }, { firstName:"James", lastName:"Bond"}]
+should convert to ["Jane Doe", "James Bond"]
+Note the space in between first and last names.
+You can assume that neither firstName nor lastName will be blank
+------------------------------------------------------------------------------------------------ */
+const toLastNames = people => {
+  return people.map(fullName => {
+    return fullName.firstName + " " + fullName.lastName;
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that, given an array of numbers as input, uses reduce to add the values in the array.
+
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr) => {
+  return arr.reduce((arr, crr) => {
+    return arr = arr + crr;
+  },0);
```

Comment:
Your naming conventions here make this a little more difficult to read.
```js
arr.reduce((arr, crr) => {
```
This is not terribly clear. You're using a argument named "arr" inside of your reduce prototype, which you're executing on the variable "arr". Also crr isn't exactly clear either.

I've seen a couple naming strategies for reduce that I like. One is to just always use (a, b)
```js
arr.reduce((a, b) => {
```
Just as a catch all.

The one I use most often is
```js
arr.reduce((answerSoFar, current) => {
```